flume-ng-core/src/main/java/org/apache/flume/source/StressSource.java

```diff
@@ -123,6 +132,9 @@ protected Status doProcess() throws EventDeliveryException {
       lastSent = batchSize;
 
       if (batchSize == 1) {
+        if (limiter != null) {
```

Comment:
Is it intentional to not having a rate limit with batchSize larger than 1?
`limiter.acquire(batchSize)` would be kind of what we need if I'm right

Reply:
That is a VERY good spot - thanks @szaboferee for being awake when reviewing!!!!!!

It's also slightly trickier than batchSize because it might be the last batch, although perhaps I didn't need to make it quite that complex. Anyhow, done.